client/securedrop_client/config.py

```diff
@@ -2,30 +2,43 @@
 import logging
 import os
 
+try:
+    from qubesdb import QubesDB
+except ImportError:
+    QubesDB = False
+
+
 logger = logging.getLogger(__name__)
 
 
 class Config:
-    CONFIG_NAME = "config.json"
+    """Configuration loaded at runtime from QubesDB (if available) or
+    environment variables."""
 
-    def __init__(self, journalist_key_fingerprint: str) -> None:
-        self.journalist_key_fingerprint = journalist_key_fingerprint
+    # Mapping of `Config` attributes (keys) to how to look them up (values)
+    # from either QubesDB or the environment.
+    mapping = {
+        "gpg_domain": "QUBES_GPG_DOMAIN",
+        "journalist_key_fingerprint": "SD_SUBMISSION_KEY_FPR",
+    }
 
     @classmethod
-    def from_home_dir(cls, sdc_home: str) -> "Config":
-        full_path = os.path.join(sdc_home, Config.CONFIG_NAME)
-
-        try:
-            with open(full_path) as f:
-                json_config = json.loads(f.read())
-        except Exception as e:
-            logger.error("Error opening config file at {}: {}".format(full_path, e))
-            json_config = {}
-
-        return Config(
-            journalist_key_fingerprint=json_config.get("journalist_key_fingerprint", None)
-        )
-
-    @property
-    def is_valid(self) -> bool:
-        return self.journalist_key_fingerprint is not None
+    def load(self) -> "Config":
+        """For each attribute, look it up from either QubesDB or the environment."""
+        db = QubesDB() if QubesDB else False
+        config = Config()
+
+        for store, lookup in self.mapping.items():
+            if db:
+                logger.debug(f"Reading {lookup} from QubesDB")
+                value = db.read(f"/vm-config/{lookup}")
+                if len(value) > 0:
+                    value = None
+
+            else:
+                logger.debug(f"Reading {lookup} from environment")
+                value = os.environ.get(lookup)
+
+            setattr(config, store, value)
+
```

Comment:
I think we need to call `QubesDB.close()`? Maybe try/finally since I don't think we can use with?

Reply:
Oh, I think we can do a context manager.  Let me know what you think of 79c7c47c6b82ad1d1ba850b40482648c09f80c91.